File: CS50x/sentimental-readability/readability.py
# TODO
from cs50 import get_string


# count_letters
def count_letters(text):
    len_text = len(text)
    letters = 0
    for i in range(len_text):
        if text[i].isalpha():
            letters = letters+1
    return letters


# count words by spaces
def count_words(text):
    len_text = len(text)
    if (len_text == 0):
        return 0
    else:
        words = 1
        for i in range(len_text):
            if text[i] == " ":
                words = words+1
        return words


# count sentences by points
def count_sentences(text):
    len_text = len(text)
    sentences = 0
    for i in range(len_text):
        if(text[i] == "." or text[i] == "?" or text[i] == '!'):
            sentences = sentences + 1
    return sentences


# main
def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    l = 100 * letters / words
    s = 100 * sentences / words
    # average
    index = round((0.0588 * l) - (0.296 * s) - 15.8)
    if (index < 1):
        print("Before Grade 1\n")
    elif (index >= 16):
        print("Grade 16+\n")
    else:
        print("Grade "+str(index)+"\n")


main()

File: CS50x/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>
//prototypes
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //get the string
    string text = get_string("Text: ");
    //words
    int letters = count_letters(text);
    float words = count_words(text);
    int sentences = count_sentences(text);
    //calc l
    float l = (letters / words) * 100;
    //calc s
    float s = (sentences / words) * 100;
    //calc index
    float index = round((0.0588 * l) - (0.296 * s) - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) index);
    }
}

//count sentences
int count_sentences(string text)
{
    int len = strlen(text);
    int sentences = 0;
    for (int i = 0 ; i < len ; i++)
    {
        //points for sentences
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    return sentences;
}
//count_words
int count_words(string text)
{
    int len = strlen(text);
    if (len == 0)
    {
        return 0;
    }
    else
    {
        int words = 1;
        for (int i = 0 ; i < len ; i++)
        {
            if (text[i] == 32)
            {
                words++;
            }
        }
        return words;
    }
}
//count letters
int count_letters(string text)
{
    int len = strlen(text);
    int letters = 0;
    for (int i = 0; i < len ; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}